llvm/lib/Analysis/ValueTracking.cpp

```diff
@@ -7318,6 +7318,25 @@ bool llvm::isOverflowIntrinsicNoWrap(const WithOverflowInst *WO,
 
 /// Shifts return poison if shiftwidth is larger than the bitwidth.
 static bool shiftAmountKnownInRange(const Value *ShiftAmount) {
+  auto *A = dyn_cast<Argument>(ShiftAmount);
+  if (A) {
+    auto Range = A->getRange();
+    if (Range && Range->getUnsignedMax().ult(
+                     ShiftAmount->getType()->getIntegerBitWidth())) {
```

Comment:
This should use getScalarSizeInBits(), otherwise it will crash for vector shifts. Same below.